[BUG] Latest bugfix commit breaks emergency parser

<!--

Please follow the instructions below. Failure to do so may result in your issue being closed.

### Before Reporting a Bug

1. Test with the `bugfix-2.0.x` branch to see whether the issue still exists.

2. Get troubleshooting help from the Marlin community to confirm it's a bug and not just a configuration error. Links at https://github.com/MarlinFirmware/Marlin/issues/new/choose

### Instructions

1. Fill out every section of the template below.

2. Always attach configuration files, regardless of whether you think they are involved.

3. Read and understand Marlin's Code of Conduct. By filing an Issue, you are expected to comply with it, including treating everyone with respect: https://github.com/MarlinFirmware/Marlin/blob/master/.github/code_of_conduct.md

-->

### Bug Description

I just updated my bugfix branch to the very latest commit (7bbdbcfb6db4c6d35e4aaffca06073dd2bfc6719) and I tried compiling the firmware and immediately got this error

```
Marlin/src/HAL/STM32/MarlinSerial.cpp: In member function 'void MarlinSerial::_rx_complete_irq(serial_t*)':
Marlin/src/HAL/STM32/MarlinSerial.cpp:84:31: error: 'emergency_state' was not declared in this scope; did you mean 'emergency_parser'?
   84 |       emergency_parser.update(emergency_state, c);
      |                               ^~~~~~~~~~~~~~~
      |                               emergency_parser
Compiling .pio/build/BIGTREE_SKR_PRO/src/src/core/utility.cpp.o
*** [.pio/build/BIGTREE_SKR_PRO/src/src/HAL/STM32/MarlinSerial.cpp.o] Error 1
```
I do have emergency paraser enabled.. 

### Configuration Files

Here's a link to my branch: https://github.com/NovaViper/Marlin/tree/xavier-stable2.0.x/Marlin
Aswell as the run: https://github.com/NovaViper/Marlin/runs/1785661931

### Steps to Reproduce

<!-- Describe the steps needed to reproduce the issue. (You can remove this invisible comment.) -->

1. Get the 7bbdbcf commit
2. Enable Emergency parser 
3. Compile for an SKR_Pro board
4. The error occurs

**Expected behavior:**

It should compile like normal and throw out the error

**Actual behavior:**

It throws out that error mentioned above

#### Additional Information

* Provide pictures or links to videos that clearly demonstrate the issue.
* See [Contributing to Marlin](https://github.com/MarlinFirmware/Marlin/blob/2.0.x/.github/contributing.md) for additional guidelines.

--- a/Marlin/src/HAL/STM32/MarlinSerial.cpp
+++ b/Marlin/src/HAL/STM32/MarlinSerial.cpp
@@ -81,7 +81,7 @@ void MarlinSerial::_rx_complete_irq(serial_t *obj) {
     }
 
     #if ENABLED(EMERGENCY_PARSER)
-      emergency_parser.update(emergency_state, c);
+      emergency_parser.update(static_cast<MSerialT*>(this)->emergency_state, c);
     #endif
   }
 }

--- a/Marlin/src/MarlinCore.cpp
+++ b/Marlin/src/MarlinCore.cpp
@@ -614,8 +614,8 @@ inline void manage_inactivity(const bool ignore_stepper_queue=false) {
  */
 void idle(TERN_(ADVANCED_PAUSE_FEATURE, bool no_stepper_sleep/*=false*/)) {
   #if ENABLED(MARLIN_DEV_MODE)
-    static uint8_t idle_depth = 0;
-    if (++idle_depth > 5) SERIAL_ECHOLNPAIR("idle() call depth: ", int(idle_depth));
+    static uint16_t idle_depth = 0;
+    if (++idle_depth > 5) SERIAL_ECHOLNPAIR("idle() call depth: ", idle_depth);
   #endif
 
   // Core Marlin activities

--- a/Marlin/src/core/serial_base.h
+++ b/Marlin/src/core/serial_base.h
@@ -78,13 +78,13 @@ struct SerialBase {
   FORCE_INLINE void write(const char* str)                    { while (*str) write(*str++); }
   FORCE_INLINE void write(const uint8_t* buffer, size_t size) { while (size--) write(*buffer++); }
   FORCE_INLINE void print(const char* str)                    { write(str); }
-  NO_INLINE void print(char c, int base = 0)               { print((long)c, base); }
-  NO_INLINE void print(unsigned char c, int base = 0)      { print((unsigned long)c, base); }
-  NO_INLINE void print(int c, int base = DEC)              { print((long)c, base); }
-  NO_INLINE void print(unsigned int c, int base = DEC)     { print((unsigned long)c, base); }
-  void print(long c, int base = DEC)            { if (!base) write(c); write((const uint8_t*)"-", c < 0); printNumber(c < 0 ? -c : c, base); }
-  void print(unsigned long c, int base = DEC)   { printNumber(c, base); }
-  void print(double c, int digits = 2)          { printFloat(c, digits); }
+  NO_INLINE void print(char c, int base = 0)              { print((long)c, base); }
+  NO_INLINE void print(unsigned char c, int base = 0)     { print((unsigned long)c, base); }
+  NO_INLINE void print(int c, int base = DEC)             { print((long)c, base); }
+  NO_INLINE void print(unsigned int c, int base = DEC)    { print((unsigned long)c, base); }
+  void print(long c, int base = DEC)                      { if (!base) write(c); else { write((const uint8_t*)"-", c < 0); printNumber(c < 0 ? -c : c, base); } }
+  void print(unsigned long c, int base = DEC)             { printNumber(c, base); }
+  void print(double c, int digits = 2)                    { printFloat(c, digits); }
 
   NO_INLINE void println(const char s[])                  { print(s); println(); }
   NO_INLINE void println(char c, int base = 0)            { print(c, base); println(); }
@@ -98,6 +98,10 @@ struct SerialBase {
 
   // Print a number with the given base
   void printNumber(unsigned long n, const uint8_t base) {
+    if (!base) {
+      write((uint8_t)n);
+      return;
+    }
     if (n) {
       unsigned char buf[8 * sizeof(long)]; // Enough space for base 2
       int8_t i = 0;

--- a/Marlin/src/core/serial_base.h
+++ b/Marlin/src/core/serial_base.h
@@ -82,9 +82,18 @@ struct SerialBase {
   NO_INLINE void print(unsigned char c, int base = 0)     { print((unsigned long)c, base); }
   NO_INLINE void print(int c, int base = DEC)             { print((long)c, base); }
   NO_INLINE void print(unsigned int c, int base = DEC)    { print((unsigned long)c, base); }
-  void print(long c, int base = DEC)                      { if (!base) write(c); else { write((const uint8_t*)"-", c < 0); printNumber(c < 0 ? -c : c, base); } }
   void print(unsigned long c, int base = DEC)             { printNumber(c, base); }
   void print(double c, int digits = 2)                    { printFloat(c, digits); }
+  void print(long c, int base = DEC)                      {
+    if (!base) {
+      write(c);
+      return;
+    }
+    if (base == DEC && c < 0) {
+      write((uint8_t)'-'); c = -c;
+    }
+    printNumber(c, base);
+  }
 
   NO_INLINE void println(const char s[])                  { print(s); println(); }
   NO_INLINE void println(char c, int base = 0)            { print(c, base); println(); }

--- a/Marlin/src/feature/bedlevel/ubl/ubl.cpp
+++ b/Marlin/src/feature/bedlevel/ubl/ubl.cpp
@@ -150,7 +150,7 @@
     SERIAL_ECHO_SP(7);
     LOOP_L_N(i, GRID_MAX_POINTS_X) {
       if (i < 10) SERIAL_CHAR(' ');
-      SERIAL_ECHO(i);
+      SERIAL_ECHO((int)i); //PLEASE LEAVE THIS COMMENT, THIS CODE IS HARD TO FIND, I'LL REVERT IT LATER ON
       SERIAL_ECHO_SP(sp);
     }
     serial_delay(10);

--- a/Marlin/src/feature/bedlevel/ubl/ubl.cpp
+++ b/Marlin/src/feature/bedlevel/ubl/ubl.cpp
@@ -150,7 +150,7 @@
     SERIAL_ECHO_SP(7);
     LOOP_L_N(i, GRID_MAX_POINTS_X) {
       if (i < 10) SERIAL_CHAR(' ');
-      SERIAL_ECHO((int)i); //PLEASE LEAVE THIS COMMENT, THIS CODE IS HARD TO FIND, I'LL REVERT IT LATER ON
+      SERIAL_ECHO((int)i);
       SERIAL_ECHO_SP(sp);
     }
     serial_delay(10);

--- a/Marlin/src/gcode/calibrate/M48.cpp
+++ b/Marlin/src/gcode/calibrate/M48.cpp
@@ -240,7 +240,7 @@ void GcodeSuite::M48() {
       sigma = SQRT(dev_sum / (n + 1));
 
       if (verbose_level > 1) {
-        SERIAL_ECHO((int)(n + 1)); //PLEASE LEAVE THIS COMMENT, THIS CODE IS HARD TO FIND, I'LL REVERT IT LATER ON
+        SERIAL_ECHO((int)(n + 1));
         SERIAL_ECHOPAIR(" of ", (int)n_samples);
         SERIAL_ECHOPAIR_F(": z: ", pz, 3);
         SERIAL_CHAR(' ');

--- a/Marlin/src/gcode/calibrate/M48.cpp
+++ b/Marlin/src/gcode/calibrate/M48.cpp
@@ -240,8 +240,8 @@ void GcodeSuite::M48() {
       sigma = SQRT(dev_sum / (n + 1));
 
       if (verbose_level > 1) {
-        SERIAL_ECHO(n + 1);
-        SERIAL_ECHOPAIR(" of ", int(n_samples));
+        SERIAL_ECHO((int)(n + 1)); //PLEASE LEAVE THIS COMMENT, THIS CODE IS HARD TO FIND, I'LL REVERT IT LATER ON
+        SERIAL_ECHOPAIR(" of ", (int)n_samples);
         SERIAL_ECHOPAIR_F(": z: ", pz, 3);
         SERIAL_CHAR(' ');
         dev_report(verbose_level > 2, mean, sigma, min, max);

--- a/Marlin/src/gcode/host/M115.cpp
+++ b/Marlin/src/gcode/host/M115.cpp
@@ -35,8 +35,8 @@
   static void cap_line(PGM_P const name, bool ena=false) {
     SERIAL_ECHOPGM("Cap:");
     serialprintPGM(name);
-    SERIAL_CHAR(':');
-    SERIAL_ECHOLN(int(ena ? 1 : 0));
+    SERIAL_CHAR(':', ena ? '1' : '0');
+    SERIAL_EOL();
   }
 #endif
 

--- a/Marlin/src/module/temperature.cpp
+++ b/Marlin/src/module/temperature.cpp
@@ -2062,7 +2062,7 @@ void Temperature::init() {
       switch (heater_id) {
         case H_BED:     SERIAL_ECHOPGM("bed"); break;
         case H_CHAMBER: SERIAL_ECHOPGM("chamber"); break;
-        default:        SERIAL_ECHO(heater_id);
+        default:        SERIAL_ECHO((int)heater_id); //PLEASE LEAVE THIS COMMENT, THIS CODE IS HARD TO FIND, I'LL REVERT IT LATER ON
       }
       SERIAL_ECHOLNPAIR(
         " ; sizeof(running_temp):", sizeof(running_temp),

--- a/Marlin/src/module/temperature.cpp
+++ b/Marlin/src/module/temperature.cpp
@@ -2062,7 +2062,7 @@ void Temperature::init() {
       switch (heater_id) {
         case H_BED:     SERIAL_ECHOPGM("bed"); break;
         case H_CHAMBER: SERIAL_ECHOPGM("chamber"); break;
-        default:        SERIAL_ECHO((int)heater_id); //PLEASE LEAVE THIS COMMENT, THIS CODE IS HARD TO FIND, I'LL REVERT IT LATER ON
+        default:        SERIAL_ECHO((int)heater_id);
       }
       SERIAL_ECHOLNPAIR(
         " ; sizeof(running_temp):", sizeof(running_temp),

--- a/Marlin/src/module/tool_change.cpp
+++ b/Marlin/src/module/tool_change.cpp
@@ -709,7 +709,7 @@ inline void fast_line_to_current(const AxisEnum fr_axis) { _line_to_current(fr_a
 #if EXTRUDERS
   inline void invalid_extruder_error(const uint8_t e) {
     SERIAL_ECHO_START();
-    SERIAL_CHAR('T'); SERIAL_ECHO((int)e); //PLEASE LEAVE THIS COMMENT, THIS CODE IS HARD TO FIND, I'LL REVERT IT LATER ON
+    SERIAL_CHAR('T'); SERIAL_ECHO((int)e);
     SERIAL_CHAR(' '); SERIAL_ECHOLNPGM(STR_INVALID_EXTRUDER);
   }
 #endif

--- a/Marlin/src/module/tool_change.cpp
+++ b/Marlin/src/module/tool_change.cpp
@@ -709,7 +709,7 @@ inline void fast_line_to_current(const AxisEnum fr_axis) { _line_to_current(fr_a
 #if EXTRUDERS
   inline void invalid_extruder_error(const uint8_t e) {
     SERIAL_ECHO_START();
-    SERIAL_CHAR('T'); SERIAL_ECHO(int(e));
+    SERIAL_CHAR('T'); SERIAL_ECHO((int)e); //PLEASE LEAVE THIS COMMENT, THIS CODE IS HARD TO FIND, I'LL REVERT IT LATER ON
     SERIAL_CHAR(' '); SERIAL_ECHOLNPGM(STR_INVALID_EXTRUDER);
   }
 #endif

--- a/Marlin/src/sd/SdBaseFile.cpp
+++ b/Marlin/src/sd/SdBaseFile.cpp
@@ -926,7 +926,7 @@ int SdBaseFile::peek() {
 // print uint8_t with width 2
 static void print2u(const uint8_t v) {
   if (v < 10) SERIAL_CHAR('0');
-  SERIAL_ECHO((int)v); //PLEASE LEAVE THIS COMMENT, THIS CODE IS HARD TO FIND, I'LL REVERT IT LATER ON
+  SERIAL_ECHO((int)v);
 }
 
 /**

--- a/Marlin/src/sd/SdBaseFile.cpp
+++ b/Marlin/src/sd/SdBaseFile.cpp
@@ -926,7 +926,7 @@ int SdBaseFile::peek() {
 // print uint8_t with width 2
 static void print2u(const uint8_t v) {
   if (v < 10) SERIAL_CHAR('0');
-  SERIAL_ECHO(int(v));
+  SERIAL_ECHO((int)v); //PLEASE LEAVE THIS COMMENT, THIS CODE IS HARD TO FIND, I'LL REVERT IT LATER ON
 }
 
 /**

